### PR TITLE
Feat on editdata item

### PR DESCRIPTION
**Pull request by EdmondCShaffer**
Closes #3911

**Summary**
Reworked the on click handler and add new prop onEditDataItem to support v2 api call to get downsample methods 

**Change List (commits, features, bugs, etc)**

- card editor 

**Acceptance Test (how to verify the PR)**

- on edit should look for downsample methods when V2

**Regression Test (how to make sure this PR doesn't break old functionality)**

- tests here

**Things to look for during review**

- [ ] Make sure all references to `iot` or `bx` class prefix is using the prefix variable
- [ ] (React) All major areas have a `data-testid` attribute. New test ids should have test written to ensure they are not changed or removed.
- [ ] UI should be checked in RTL mode to ensure the proper handling of layout and text.
- [ ] All strings should be translatable.
- [ ] The code should pass a11y scans (The storybook a11y knob should show no violations). New components should have a11y test written.
- [ ] Unit test should be written and should have a coverage of 90% or higher in all areas.
- [ ] All components should be passing visual regression test. For new styles or components either a visual regression test should be written for all permutations or the base image updated.
- [ ] Changes or new components should either write new or update existing documentation.
- [ ] PR should link and close out an existing issue

### CODE REVIEW

**Review by davidicus — APPROVED** [review]

File: packages/react/src/components/HotspotEditorModal/HotspotEditorDataSourceTab/HotspotEditorDataSourceTab.test.jsx [hunk 1]
```diff
@@ -146,19 +163,44 @@ describe('HotspotEditorDataSourceTab', () => {
...
+        hotspot={cardConfigWithPresets.content.hotspots[0]}
+        cardConfig={cardConfigWithPresets}
+        dataItems={dataItemsV2}
+        onChange={onChange}
+        translateWithId={() => {}}
+        onEditDataItem={onEditDataItem}
+      />
+    );
+    userEvent.click(screen.getAllByRole('button')[2]);
+    await waitFor(() => expect(screen.getByRole('dialog')).toBeInTheDocument());
```
Comment (davidicus) [inline packages/react/src/components/HotspotEditorModal/HotspotEditorDataSourceTab/HotspotEditorDataSourceTab.test.jsx:202]: Isn't this the same assertion that we have on 204?

**Review by davidicus — COMMENTED** [review]
just the one comment about the superfluous test.

**Review by davidicus — CHANGES_REQUESTED** [review]
just the one comment about test and we should be able to merge

File: packages/react/src/components/HotspotEditorModal/HotspotEditorDataSourceTab/HotspotEditorDataSourceTab.test.jsx (hunk 1 above)
Comment (EdmondCShaffer) [reply packages/react/src/components/HotspotEditorModal/HotspotEditorDataSourceTab/HotspotEditorDataSourceTab.test.jsx:200]: Yes I removed 204 and 186

**Review by davidicus — APPROVED** [review]

**Pull request merged**